src/dotnet/ToolPackage/LockFileMatcher.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class LockFileMatcher
+    {
+        private static string[] _pathInLockFilePathInArray;
+        private static string[] _entryPointPathInArray;
+
+        internal static bool MatchesFile(string pathInLockFile, string entryPoint)
+        {
+            _pathInLockFilePathInArray = SplitPathByDirectorySeparator(pathInLockFile);
+            _entryPointPathInArray = SplitPathByDirectorySeparator(entryPoint);
+
+            return EntryPointHasFileAtLeast()
+                   && PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
+                   && RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories();
+        }
+
+        private static bool RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories()
+        {
+            string[] pathAfterToolsTfmRid = _pathInLockFilePathInArray.Skip(3).ToArray();
+            return !pathAfterToolsTfmRid
+                .Where((directoryOnEveryLevel, i) => directoryOnEveryLevel != _entryPointPathInArray[i])
+                .Any();
+        }
+
+        private static bool PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
+        {
+            if (_pathInLockFilePathInArray.Length - _entryPointPathInArray.Length != 3)
+            {
+                return false;
+            }
+
+            if (_pathInLockFilePathInArray[0] != "tools")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool EntryPointHasFileAtLeast()
+        {
+            return _entryPointPathInArray.Length >= 1;
```

Comment:
Do we really need a method for checking if `_entryPointPathInArray` is empty?

Reply:
`return EntryPointHasFileAtLeast()
                  && PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
                   && RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories(); `

If we do not check it there, PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories will throw. This 3 check is more and more specific

Reply:
Sorry, I didn't mean to imply that the check itself is unneeded, only refactoring the check out into its own method when it is only checked from exactly one place.  Every time a method call is introduced I now have to stop and jump to the method's definition to see what it does.  It would be far more concise to simply do a check for empty where needed.

Reply:
I think Peter is not saying don't check it, he's just saying it doesn't seem worth it to factor the length check out into a separate method.

Reply:
will in line this method